logs_sdk/lib/opentelemetry/sdk/logs/logger_provider.rb

```diff
@@ -0,0 +1,129 @@
+          end
+        end
+
+        # Attempts to stop all the activity for this {LoggerProvider}. Calls
+        # LogRecordProcessor#shutdown for all registered LogRecordProcessors.
+        #
+        # This operation may block until all the Log Records are processed. Must
+        # be called before turning off the main application to ensure all data
+        # are processed and exported.
+        #
+        # After this is called all the newly created {LogRecord}s will be no-op.
+        #
+        # @param [optional Numeric] timeout An optional timeout in seconds.
+        # @return [Integer] Export::SUCCESS if no error occurred, Export::FAILURE if
+        #   a non-specific failure occurred, Export::TIMEOUT if a timeout occurred.
+        def shutdown(timeout: nil)
+          @mutex.synchronize do
+            if @stopped
+              OpenTelemetry.logger.warn('LoggerProvider#shutdown called multiple times.')
+              return OpenTelemetry::SDK::Logs::Export::FAILURE
+            end
+
+            start_time = OpenTelemetry::Common::Utilities.timeout_timestamp
+            results = @log_record_processors.map do |processor|
+              remaining_timeout = OpenTelemetry::Common::Utilities.maybe_timeout(timeout, start_time)
+              break [OpenTelemetry::SDK::Logs::Export::TIMEOUT] if remaining_timeout&.zero?
+
+              processor.shutdown(timeout: remaining_timeout)
+            end
+
+            @stopped = true
+            results.max || OpenTelemetry::SDK::Logs::Export::SUCCESS
+          end
+        end
+
+        # Immediately export all log records that have not yet been exported
+        # for all the registered LogRecordProcessors.
+        #
+        # This method should only be called in cases where it is absolutely
+        # necessary, such as when using some FaaS providers that may suspend
+        # the process after an invocation, but before the `Processor` exports
+        # the completed log records.
+        #
+        # @param [optional Numeric] timeout An optional timeout in seconds.
+        # @return [Integer] Export::SUCCESS if no error occurred, Export::FAILURE if
+        #   a non-specific failure occurred, Export::TIMEOUT if a timeout occurred.
+        def force_flush(timeout: nil)
+          @mutex.synchronize do
+            return Export::SUCCESS if @stopped
+
+            start_time = OpenTelemetry::Common::Utilities.timeout_timestamp
+            results = @log_record_processors.map do |processor|
+              remaining_timeout = OpenTelemetry::Common::Utilities.maybe_timeout(timeout, start_time)
+              return Export::TIMEOUT if remaining_timeout&.zero?
+
+              processor.force_flush(timeout: remaining_timeout)
+            end
+
+            results.max || Export::SUCCESS
+          end
+        rescue StandardError => e
+          OpenTelemetry.handle_error(exception: e, message: FORCE_FLUSH_ERROR)
+          Export::FAILURE
+        end
```

Comment:
There is not a `rescue` in the [TracerProvider#force_flush method](https://github.com/open-telemetry/opentelemetry-ruby/blob/4347e0f47bb0d6a4870f1e9311f20d3d344fb34a/sdk/lib/opentelemetry/sdk/trace/tracer_provider.rb#L89-L113), but the method documentation states "`Export::FAILURE` should be returned if a non-specific failure occurred." I'm assuming a non-specific failure would be any `StandardError` instance. 

* Is there a condition to return `Export::FAILURE` on non-specific errors in `TracerProvider#force_flush` that I'm missing? 
* Similar documentation exists for the `LoggerProvider`/`TracerProvider#shutdown` methods. If we choose to add a `rescue` here, should the `#shutdown` methods also have a similar `rescue`?